File: server/src/repositories/measurement-repository.ts
import type { Measurement, NewMeasurement } from "../domain/measurement.js";

// Маршруты зависят от контракта, а не от способа хранения. Реализацию можно
// заменить на PostgreSQL-репозиторий без изменений в API.
export interface MeasurementRepository {
  findAll(): Promise<Measurement[]>;
  create(measurement: NewMeasurement): Promise<Measurement>;
}

File: internal/repository/measurement_repository.go
package repository

import (
	"context"

	"github.com/devldavydov/myhealth-ng/internal/domain"
)

type MeasurementRepository interface {
	FindAll(context.Context) ([]domain.Measurement, error)
	Create(context.Context, domain.NewMeasurement) (domain.Measurement, error)
}
